frameworks/kafka/src/main/java/com/mesosphere/sdk/kafka/upgrade/CuratorStateStoreFilter.java

```diff
@@ -0,0 +1,60 @@
+package com.mesosphere.sdk.kafka.upgrade;
+
+import com.mesosphere.sdk.curator.CuratorStateStore;
+import com.mesosphere.sdk.curator.CuratorUtils;
+import com.mesosphere.sdk.offer.evaluate.placement.StringMatcher;
+import com.mesosphere.sdk.state.StateStoreException;
+import org.apache.curator.RetryPolicy;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * CuratorStateStoreFilter enables to ignore task names that match the filter.
+ */
+public class CuratorStateStoreFilter  extends CuratorStateStore {
```

Comment:
Is this class really needed ? Adding a class adds a lot of complexity to code-base. And, in case of this class, it's not doing much. I would suggest you to please re-think about adding this class.